sbi/samplers/rejection/rejection.py

```diff
@@ -269,19 +275,26 @@ def accept_reject_sample(
             (sampling_batch_size,),  # type: ignore
             **proposal_sampling_kwargs,
         )
-
         # SNPE-style rejection-sampling when the proposal is the neural net.
         are_accepted = accept_reject_fn(candidates)
-        samples = candidates[are_accepted]
-        accepted.append(samples)
+        # Reshape necessary in certain cases which do not follow the shape conventions
+        # of the "DensityEstimator" class.
+        are_accepted = are_accepted.reshape(sampling_batch_size, num_xos)
+        candidates_to_reject = candidates.reshape(
+            sampling_batch_size, num_xos, *candidates.shape[candidates.ndim - 1 :]
+        )
+
+        num_accepted = are_accepted.sum(dim=0).min().item()
```

Comment:
Yeah this would be better. In a vectorized setting this does not matter to much, I would not do this in this PR (but one could keep it in mind, one would have to by mask out the conditions that are finished).